### ISSUE TITLE
Evaluation protocol & Pretrained Models

### ISSUE THREAD

**Issue by elias-ramzi** [SupetZYK/DynamicMetricLearning#4]
Hi !

I am working with your datasets.

I have some more questions, do you plan on releasing :

- Training protocol (optimizer, pretrain or not resnet34, mining code...)
- Hyperparameters used in your paper (ie. learning rates, hyperparameters for the loss etc.)
- The evaluation protocol / scripts ?
- Pre-trained models ?

Thanks !

**Comment by DavidLeexxxx** [comment]
Hi, I also have some kinds of problem about the impelementation details. Since the DSL loss concerns the within-class similarity from different scales, so what is the batch sampling strategy for this method when training with DSL loss by using a mini-batch gradient descent manner. Apparently sample a batch by random shuffle is not suitable here.

**Comment by elias-ramzi** [comment]
Hi, I have worked with the CSL loss and used the same sampling as in standard image retrieval (e.g. m-per-class sampler).
You can find an implementation of the loss here : https://github.com/elias-ramzi/HAPPIER/blob/main/happier/losses/csl_loss.py
Hope this helps.

**Comment by DavidLeexxxx** [comment]
Thanks for your reply and the future work HAPPIER,I will keep learning and focusing on them.
